src/v/crypto/key.cc

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "key.h"
+
+#include "crypto/crypto.h"
+#include "crypto/exceptions.h"
+#include "crypto/ssl_utils.h"
+#include "crypto/types.h"
+
+#include <absl/container/flat_hash_set.h>
+#include <openssl/bio.h>
+#include <openssl/bn.h>
+#include <openssl/evp.h>
+#include <openssl/param_build.h>
+#include <openssl/pem.h>
+
+namespace crypto {
+
+namespace internal {
+EVP_PKEY_ptr load_public_pem_key(bytes_view key) {
+    BIO_ptr key_bio(BIO_new_mem_buf(key.data(), key.size()));
+    auto pkey_temp = EVP_PKEY_new();
+    if (
+      nullptr
+      == PEM_read_bio_PUBKEY(key_bio.get(), &pkey_temp, nullptr, nullptr)) {
+        EVP_PKEY_free(pkey_temp);
+        throw ossl_error("Failed to load PEM public key");
+    }
+
+    return EVP_PKEY_ptr(pkey_temp);
```

Comment:
I'm probably misreading how this works, but how about:
```suggestion
    EVP_PKEY_ptr pkey(
      PEM_read_bio_PUBKEY(key_bio.get(), nullptr, nullptr, nullptr));
    if (!pkey) {
        throw ossl_error("Failed to load PEM public key");
    }

    return pkey;
```

Reply:
No I think you're right you can do it that way... I've always done it this way but maybe the API changed a while ago and I got stuck in my way of doing it in this pattern...